DOCUMENTATION.md

```diff
@@ -0,0 +1,99 @@
+
+Vampire poison sting (card that doesn't exist btw): Deal 1 damage to all enemies, inflict 3 poison to the target and heal yourself for 2 HP.
+
+All the `EffectData` (there are 3 here) are held inside the `card_effects_data` array of the card. The card will then loop through the array and play the effects one by one, ie an `EffectDamage`, an `EffectPoison` and an `EffectHeal` on the relevant targets.
+
+## Targeting system
+
+WiP
+
+## Status
+
+`Status` is the way we deal with conditions on a character that stays on them for multiple turns. For example, poison, weakness, strength... There are 2 catergories of `Status`, which are `Buff` and `Debuff`. The difference between effects and status is that effects do something when a card is played and status do something at a given time in the turn. The `EffectPoison` applies a `DebuffPoison` to the target.
+
+Each `Entity` has an array of Status where we store said `Status`. At the start of the turn of each entity, the `Status` inside the array are applied (and the number of turns left is decreased by 1, if the status is not for the entire duration of the fight). Note that conditions that carry between fights (such as conditions provided by relics) do not qualify as `Status`, they are handled differently.
+
+# Entity
+
+# Map
+
+The structure of the map is the following:
+
+![Map structure](documentation_images/map_concept.png)
+
+## Rooms
+
+Rooms are the basis of the map (they are the squares you can see on the image of the map). The map is a 2D Array of rooms. Each room holds basic information about the event in this room as well as informations about the presence of a light source and the level of light in the room (see [Light mechanic](#light-mechanic) for more informations).
+
+## Events
+
+What might happen in this new room ? Mystery, mystery... Events is the way we decide what is going to happen in each room. Currently, the list of events is the following:
+
+- Mob: an encounter with enemies
+- Random: you don't know what is going to happen. Most likely, you will face a unique scenario with choices, but it might also be a mob, a shop, a heal or any other event.
+- Shop: buy and sell cards
+- Heal: well... as it says
+
+Events can change depending on the level of light in the room. The distribution of events is decided at map level to have something coherent.
+
+## Moving on the map
+
+Players will move once the event in the room they are in is over. They must go up one floor when going in a new room. The player can go to the room directly above them or the ones on the sides. This movement pattern may change due to some relics.
+
+## Light mechanic
+
+One of the main feature of the game is the ability for the player to influence the map through his actions. The player can place a light source (refered as "torch" for now) in a room, after the event is over and before using the map. The light source will then be used to light the room and the rooms around it.
+
```

Comment:
Maybe include a paragraph which includes what light does in more concrete terms before getting to the example.

Reply:
I'm a bit confused, isn't an example the most concrete way to describe something ?

Reply:
I feel like having a more dry explenation on what something does is good scince it's faster to read, and less interpretaion is required, then haveing an example can clear up what may be confusing in the initial expelation.

Reply:
i see, so it's more contextual and less concrete that you want then ?

Reply:
It's more that I want just a quick explantion as something like a tldr or something

Reply:
Alright